api/src/wfl/api/spec.clj

```diff
@@ -66,40 +96,39 @@
                                             ::uuid
                                             ::version]))
 (s/def ::workload-responses (s/* ::workload-response))
+(s/def ::workspace string?)
 
 ;; compound
 (s/def ::items (s/* ::workload-inputs))
-(s/def ::workload-inputs (s/keys :req-un [::inputs]
-                                 :opt-un [::options]))
 (s/def ::inputs (s/or :aou      ::aou-workflow-inputs
                       :arrays   ::arrays-workflow-inputs
                       :copyfile ::copyfile-workflow-inputs
                       :wgs      ::wgs-workflow-inputs
                       :xx       ::xx-workflow-inputs
                       :sg       ::sg-workflow-inputs))
-
-(s/def ::workflows (s/* ::workflow))
 (s/def ::workflow
   (s/keys :req-un [::inputs]
           :opt-un [::status ::updated ::uuid ::options]))
+(s/def ::workflows (s/* ::workflow))
+(s/def ::workload-inputs (s/keys :req-un [::inputs]
+                                 :opt-un [::options]))
 
 ;; aou
 (s/def ::analysis_version_number integer?)
-(s/def ::chip_well_barcode string?)
-(s/def ::append-to-aou-request (s/keys :req-un [::notifications ::uuid]))
-(s/def ::append-to-aou-response (s/* ::aou-workflow-inputs))
 (s/def ::aou-workflow-inputs (s/keys :req-un [::analysis_version_number
                                               ::chip_well_barcode]))
-
-(s/def ::notifications (s/* ::aou-sample))
 (s/def ::aou-sample (s/keys :req-un [::analysis_version_number
                                      ::chip_well_barcode]))
+(s/def ::append-to-aou-request (s/keys :req-un [::notifications ::uuid]))
+(s/def ::append-to-aou-response (s/* ::aou-workflow-inputs))
+(s/def ::chip_well_barcode string?)
+(s/def ::notifications (s/* ::aou-sample))
 
 ;; arrays
-(s/def ::entity-name string?)
-(s/def ::entity-type string?)
 (s/def ::arrays-workflow-inputs (s/keys :req-un [::entity-name
                                                  ::entity-type]))
+(s/def ::entity-name string?)
```

Comment:
You need to define these specs before they are referenced, did you move these below `::arrays-workflow-inputs` for a reason?

Reply:
No - I will revert this. I think this was done during my refactoring work of the original Rex checkin, but I understand better now how they are used, and why they were wrong. Thanks!